internal/agent/upgrade_test.go

```diff
@@ -0,0 +1,47 @@
+package agent
```

Comment:
This test fails on CI because it doesn't stub anything so it tried to mount some partition.

Reply:
I removed the test. Make this code testable requires a lot more refactoring. Might happen on another PR.